homeassistant/components/advantage_air/select.py

```diff
@@ -21,11 +21,8 @@ async def async_setup_entry(
 
     instance: AdvantageAirData = hass.data[ADVANTAGE_AIR_DOMAIN][config_entry.entry_id]
 
-    entities: list[SelectEntity] = []
     if aircons := instance.coordinator.data.get("aircons"):
-        for ac_key in aircons:
-            entities.append(AdvantageAirMyZone(instance, ac_key))
-    async_add_entities(entities)
+        async_add_entities(AdvantageAirMyZone(instance, ac_key) for ac_key in aircons)
```

Comment:
FYI this is not a list comprehension, it's a generator expression, which is an issue if the callback does anything except iterate it (once)

Reply:
`async_add_entities` can accept a generator.